File: backend/ent/schema/supplier.go
package schema

import (
	"time"

	"entgo.io/ent"
	"entgo.io/ent/schema/edge"
	"entgo.io/ent/schema/field"
	"entgo.io/ent/schema/index"
	"github.com/google/uuid"
)

type Supplier struct {
	ent.Schema
}

func (Supplier) Fields() []ent.Field {
	return []ent.Field{
		field.UUID("id", uuid.UUID{}).Default(uuid.New),
		field.String("code").Unique().NotEmpty(),
		field.String("name").NotEmpty(),
		field.JSON("contact", map[string]interface{}{}).Optional(),
		field.Bool("is_active").Default(true),
		field.Time("created_at").Default(time.Now).Immutable(),
		field.Time("updated_at").Default(time.Now).UpdateDefault(time.Now),
	}
}

func (Supplier) Edges() []ent.Edge {
	return []ent.Edge{
		edge.From("tenant", Tenant.Type).Ref("suppliers").Unique().Required(),
		edge.To("purchase_orders", PurchaseOrder.Type),
	}
}

func (Supplier) Indexes() []ent.Index {
	return []ent.Index{
		index.Fields("code"),
		index.Fields("is_active"),
		index.Edges("tenant", "code").Unique(), // Supplier code unique per tenant
	}
}

File: mobile_app/lib/models/supplier.dart
class Supplier {
  final String id;
  final String code;
  final String name;
  final Map<String, dynamic>? contact;
  final bool isActive;

  Supplier({
    required this.id,
    required this.code,
    required this.name,
    this.contact,
    required this.isActive,
  });

  factory Supplier.fromJson(Map<String, dynamic> json) {
    return Supplier(
      id: json['id'],
      code: json['code'].toString(),
      name: json['name'],
      contact: json['contact'] as Map<String, dynamic>?,
      isActive: json['is_active'] ?? true,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'code': code,
      'name': name,
      'contact': contact,
      'is_active': isActive,
    };
  }

  String get displayName => '$code - $name';

  String? get email => contact?['email'];
  String? get phone => contact?['phone'];
  String? get address => contact?['address'];
}
